pkg/virt-controller/watch/application.go

```diff
@@ -89,14 +104,58 @@ func (vca *VirtControllerApp) Run() {
 	stop := make(chan struct{})
 	defer close(stop)
 	vca.informerFactory.Start(stop)
-	go vca.vmController.Run(3, stop)
-	//FIXME when we have more than one worker, we need a lock on the VM
-	go vca.migrationController.Run(3, stop)
-	httpLogger := logger.With("service", "http")
-	httpLogger.Info().Log("action", "listening", "interface", vca.host, "port", vca.port)
-	if err := http.ListenAndServe(vca.host+":"+strconv.Itoa(vca.port), nil); err != nil {
+
+	run := func(stopCh <-chan struct{}) {
+		go vca.vmController.Run(3, stop)
+		//FIXME when we have more than one worker, we need a lock on the VM
+		go vca.migrationController.Run(3, stop)
+		httpLogger := logger.With("service", "http")
+		httpLogger.Info().Log("action", "listening", "interface", vca.host, "port", vca.port)
+		if err := http.ListenAndServe(vca.host+":"+strconv.Itoa(vca.port), nil); err != nil {
+			golog.Fatal(err)
+		}
+	}
+
+	recorder := createRecorder(vca.clientSet)
+
+	rl, err := resourcelock.New(resourcelock.EndpointsResourceLock,
+		"default",
```

Comment:
Is there a more appropriate kubevirt system related namespace for the time lease locks?

Reply:
"kube-system" is our target namespace at the end. Tracked by #300